src/main/java/com/example/ContinuousIntegrationServer.java

```diff
@@ -165,8 +167,12 @@ private void handlePushEvent(Request baseRequest,
         else
             System.out.println("[webhook] Commit " + commit.id + " state set to " + newState + "!");
 
+        // test the project
+        boolean testStatus = project.testProject(buildLogs);
+
         // push the build result to database
-        String logEntryID = database.addInfo(new BuildInfo(pushEvent.headCommit.id, buildLogs.toString(), pushEvent.headCommit.timestamp));
+        String userNames = pushEvent.commits.stream().map(c -> c.author.name).distinct().collect(Collectors.joining(","));
+        String logEntryID = database.addInfo(new BuildInfo(pushEvent.headCommit.id, userNames, pushEvent.headCommit.timestamp, buildStatus ? 1 : 0, testStatus ? 1 : 0, buildLogs.toString()));
```

Comment:
This buildInfo do not match the current DatabastHistory, maybe merge #45  into #41

Reply:
sorry i can't find the difference between the BuildInfo on issue/#45 and the one on issue/#41, can you point it out to me so that i changed the code accordingly ? 
also i had merged #41 into #45 prior to this modification so i thought everything was the same?

Reply:
Yeah i think #45 is checkout from #41, and they both have 6 columns for the database. And we need modify the DatabaseHistory (to make it display all the data, as now it only display 3). Then we can merge it into main. 
So if we keep working on branch issue/#45, we can delete branch issue/#41 now.

Reply:
we don't need to delete the branch, u can just merge issue/#45 when u're done and i'll rebase and merge issue/#41 when you're done with #45 :)

Reply:
I think Leon has assigned issue #41 to himself, I am not sure whether he would like to finish the DatabaseHistory

Reply:
my bad i mixed up both numbers. i meant when #41 is done i can rebase and merge #45 on top of it

Reply:
You guys just merge, i didn't want to do anything except adapting the page generation code. If you did that already then you can just merge and close both issues.

Reply:
i added the missing columns and did some restyling. Have a look at it and if its good you guys can merge the branches however you like. I hope i pushed to the right branch